cdap-data-fabric/src/main/java/co/cask/cdap/data/dataset/DatasetInstantiator.java

```diff
@@ -39,24 +39,20 @@
  */
 public class DatasetInstantiator implements DatasetContext {
 
-  private final DatasetFramework datasetFramework;
-  // the class loader to use for data set classes
-  private final ClassLoader classLoader;
-  private final Set<TransactionAware> txAware = Sets.newIdentityHashSet();
-  // in this collection we have only datasets initialized with getDataset() which is OK for now...
-  private final Map<TransactionAware, String> txAwareToMetricNames = Maps.newIdentityHashMap();
```

Comment:
this map was being populated, but never used in the old implementation so i just removed it.